src/main/java/com/coveo/nashorn_modules/Module.java

```diff
@@ -131,7 +132,11 @@ public Object require(String module) throws ScriptException {
 
     } finally {
       // Finally, remove successful resolves from the refCache
-      refCache.remove();
+      if(requestedFullPath != null){
+        refCache.get().remove(requestedFullPath);
+      }else{
+        refCache.remove();;
```

Comment:
Since you're essentially cleaning up in `finally` I can't really see why you'd flush the *entire* cache at this point, vs just letting other stuff (owned by code higher up in the call stack). But I haven't thought about this for hours either. I'd say that if you can come up with a scenario where this is needed (with a test to prove it :grin:) it's fine, otherwise it might not be needed.